.github/workflows/build-pr.yml

```diff
@@ -0,0 +1,84 @@
+---
+name: build-pr
+on:
+  push:
+    branches:
+      # TODO: remove the `gha-` to match the current workflow again
+      - "build-gha-pr-*"
+      - "ios-gha-pr-*"
+      - "android-gha-pr-*"
+
+env:
+  NODE_VERSION: 18.7.0
+  TERM: xterm
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/test
+        uses: ./.github/actions/test
+
+  build-ios-pr:
+    runs-on: macos-12
+    if: ${{ !startsWith(github.ref_name, 'android') }}
+    needs:
+      - test
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: ci/prepare-ios-build
+        uses: ./.github/actios/prepare-ios-build
+
+      - name: ci/build-ios-pr
+        env:
+          BRANCH_TO_BUILD: "${{ github.ref_name }}"
+          AWS_ACCESS_KEY_ID: "${{ secrets.MM_MOBILE_PR_AWS_ACCESS_KEY_ID }}"
+          AWS_SECRET_ACCESS_KEY: "${{ secrets.MM_MOBILE_PR_AWS_SECRET_ACCESS_KEY }}"
+          FASTLANE_TEAM_ID: "${{ secrets.MM_MOBILE_FASTLANE_TEAM_ID }}"
+          IOS_API_ISSUER_ID: "${{ secrets.MM_MOBILE_IOS_API_ISSUER_ID }}"
+          IOS_API_KEY: "${{ secrets.MM_MOBILE_IOS_API_KEY }}"
+          IOS_API_KEY_ID: "${{ secrets.MM_MOBILE_IOS_API_KEY_ID }}"
+          MATCH_GIT_URL: "${{ secrets.MM_MOBILE_MATCH_GIT_URL }}"
+          MATCH_PASSWORD: "${{ secrets.MM_MOBILE_MATCH_PASSWORD }}"
+          MATTERMOST_WEBHOOK_URL: "${{ secrets.MM_MOBILE_PR_MATTERMOST_WEBHOOK_URL }}"
+        run: bundle exec fastlane ios build --env ios.pr
+        working-directory: ./fastlane
+
+      - name: ci/upload-ios-pr-build
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: ios-build-pr-${{ github.run_id }}
+          path: "*.ipa"
+
+  build-android-pr:
+    runs-on: ubuntu-22.04
+    if: ${{ !startsWith(github.ref_name, 'ios') }}
+    needs:
+      - test
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: ci/prepare-android-build
+        uses: ./.github/actios/prepare-android-build
+        with:
+          sign: true
+
+      - name: ci/build-android-pr
+        env:
+          BRANCH_TO_BUILD: "${{ github.ref_name }}"
+          AWS_ACCESS_KEY_ID: "${{ secrets.MM_MOBILE_PR_AWS_ACCESS_KEY_ID }}"
+          AWS_SECRET_ACCESS_KEY: "${{ secrets.MM_MOBILE_PR_AWS_SECRET_ACCESS_KEY }}"
+          MATTERMOST_WEBHOOK_URL: "${{ secrets.MM_MOBILE_PR_MATTERMOST_WEBHOOK_URL }}"
+        run: bundle exec fastlane android build --env android.pr
+        working-directory: ./fastlane
+
+      - name: ci/upload-android-pr-build
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
```

Comment:
Here is [the code](https://github.com/actions/upload-artifact). In any case it will upload the build artifact on the Actions page . Like [this](https://github.com/mattermost/mattermost-server/actions/runs/4606160534)